src/setup_node_env/index.js

```diff
@@ -0,0 +1,3 @@
+require('./node_version_validator');
+require('./ts_node_register');
```

Comment:
Lets keep the ts-node stuff in with babel_register, with babel 7 it actually will be a part of babel_register, and I think it's important that places like the plugin helpers that are pulling in babel_register specifically get typescript support too